### PR TITLE
verification display logic for american indian status self attestation

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements:
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes/features), and they use `let` helpers and `before` blocks.
- [x] For all UI changes, there is Cucumber coverage.
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, the reasoning is documented in the PR and code.
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run them is documented in both the PR and the code.
- [ ] There are no inline styles added.
- [ ] There is no inline JavaScript added.
- [ ] There is no hard-coded text added/updated in helpers/views/JavaScript. New/updated translation strings do not include markup/styles unless there is supporting documentation.
- [ ] Code does not use `.html_safe`.
- [ ] All images added/updated have alt text.
- [x] Does not bypass RuboCop rules in any way.

# PR Type
What kind of change does this PR introduce?:

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix, Migration, or Report (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/188559336

# A brief description of the changes:

Current behavior: New consumers or brokers representing new consumers, existing consumers or brokers representing existing consumers, and admins can always see their American Indian status

New behavior:  New consumers or brokers representing new consumers will never see their American Indian status when the feature flag is on. Existing consumers or brokers representing existing consumers will only see their American Indian status with the feature flag on if they have ever had their status in outstanding state. Admins can still see all users' American Indian status, regardless of flag.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and indicate which client(s) it applies to.

Variable name: `AI_AN_SELF_ATTESTATION_IS_ENABLED`

- [ ] DC
- [x] ME

# Additional Context

Changes done in the ```FamiliesController``` are specifically to fix an issue with how the bs4 feature flag interacts with Cucumbers. For some reason, Cucumbers only recognize the bs4 flag when pulling in the stylesheets and has issues with the html.erb formatted page.

### CODE REVIEW

**Review by charlienparker — DISMISSED** [review]

File: app/helpers/verification_helper.rb [hunk 1]
```diff
@@ -50,19 +50,29 @@ def display_verification_type_name(v_type)
...
+  # or if the feature for the verification type is disabled
   def can_display_type?(verif_type)
-    return true unless verif_type.type_name == 'Alive Status'
-    return false unless EnrollRegistry.feature_enabled?(:enable_alive_status)
-    return true if current_user.has_hbx_staff_role?
-
-    previous_states = verif_type.type_history_elements.pluck(:from_validation_status).compact
-    return true if previous_states.include?('outstanding')
-    return true if verif_type.validation_status == 'outstanding'
+    return true if current_user.has_hbx_staff_role? || had_outstanding_status?(verif_type)
```
Comment (mdkaraman) [inline app/helpers/verification_helper.rb:66]: I think we need to move the `had_outstanding_status?` check into the `case` statement below.  Thinking about the following scenario:
- user is a consumer
- verification type never had outstanding status
- `ai_an_self_attestation` is disabled

Then the AI/AN verification type would not display.

I think we can change line 72 to account for this:
```
!EnrollRegistry.feature_enabled?(:ai_an_self_attestation) || had_outstanding_status?(verif_type)
```

And similar, but slightly different for Alive Status:
```
EnrollRegistry.feature_enabled?(:alive_status) && had_outstanding_status?(verif_type)
```

File: app/helpers/verification_helper.rb [hunk 2]
```diff
@@ -390,4 +400,19 @@ def has_active_resident_dependent?(person,dependent)
...
     ridp_type_status(ridp_type, person) != 'valid'
   end
+
+  private
+
+  def fetch_previous_states(verif_type)
+    verif_type&.type_history_elements&.pluck(:from_validation_status, :to_validation_status)&.flatten&.compact
+  end
+
+  def fetch_previous_states_history(verif_type, previous_states)
```
Comment (mdkaraman) [inline app/helpers/verification_helper.rb:410]: Minor nitpick in the name here.  Could be slightly misleading as it is currently.  This doesn't really fetch states from the history tracks, but checks if any are equal to "outstanding".

File: features/step_definitions/individual_steps.rb [hunk 3]
```diff
@@ -1248,6 +1248,44 @@ class VlpDocument
   visit "/families/home"
 end
 
+Then(/^.+ should (.*) an American Indian or Alaska Native status$/) do |is_outstanding|
+  if is_outstanding == 'not see'
+    expect(page).not_to have_content("American Indian Status")
```
Comment (mdkaraman) [inline features/step_definitions/individual_steps.rb:1253]: Just noting-- Ideally we wouldn't be checking for hard coded content like this, but I realize we will be overhauling the UI shortly anyways and adding a unique `id` to the existing partial is not worth the trouble.

File: features/step_definitions/individual_steps.rb [hunk 4]
```diff
@@ -1248,6 +1248,44 @@ class VlpDocument
...
+
+Given(/^the ai_an_self_attestation feature is enabled$/) do
+  allow(EnrollRegistry[:ai_an_self_attestation].feature).to receive(:is_enabled).and_return(true)
+end
+
+Given(/^the ai_an_self_attestation feature is disabled$/) do
+  allow(EnrollRegistry[:ai_an_self_attestation].feature).to receive(:is_enabled).and_return(false)
+end
+
+Given(/^the consumer is (.*) user with (.*) American Indian or Alaska Native status$/) do |existing, status|
```
Comment (mdkaraman) [inline features/step_definitions/individual_steps.rb:1267]: Great idea making this a flexible step definition!

**Review by mdkaraman — CHANGES_REQUESTED** [review]
Nice work on the new cucumbers and thorough spec coverage!  Just one important thing to check on to be sure the logic checks out, but otherwise looks good.

File: features/step_definitions/individual_steps.rb (hunk 3 above)
Comment (ATBull81) [reply features/step_definitions/individual_steps.rb]: I agree, for the moment I at least modified them to use the constants of American Indian Status: https://github.com/ideacrew/enroll/pull/4774/commits/6d80ef089df5eedc559b94e5d2342b53c0eb98c5#diff-b911f5fd6f4a943e31c8c93c7ed13793a5f223a7dd19a1d761712998594a6fdcR1253-R1255

File: app/helpers/verification_helper.rb (hunk 2 above)
Comment (ATBull81) [reply app/helpers/verification_helper.rb]: renamed to ```has_outstanding_history?```, let me know if you'd prefer anything else https://github.com/ideacrew/enroll/pull/4774/commits/6d80ef089df5eedc559b94e5d2342b53c0eb98c5#diff-aad1e1905cc0f848564bc64f88daf76e2057c834703c121490f1cfc22e815ac4R413

**Review by mdkaraman — APPROVED** [review]
Looks good!

**Review by charlienparker — APPROVED** [review]

**Pull request merged**